framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4619,452 @@ private void makeGroundTargetType(
     addDefaultAnnotations(functionalType);
   }
 
+  /**
+   * Return true if {@code type} should be captured.
+   *
+   * <p>{@code type} should be captured if:
+   *
+   * <ul>
+   *   <li>{@code type} and {@code typeMirror} are both declared types.
+   *   <li>{@code type} does not have an uninferred type argument, has a wildcard as a type argument
+   *       and was not a raw type.
+   *   <li>{@code typeMirror} has a captured type variable as a type argument.
+   * </ul>
+   *
+   * @param type annotated type that might need to be captured
+   * @param typeMirror the capture of the underlying type of {@code type}
+   * @return true if {@code type} should be captured
+   */
+  private boolean shouldCapture(AnnotatedTypeMirror type, TypeMirror typeMirror) {
+    if (type.getKind() != TypeKind.DECLARED || typeMirror.getKind() != TypeKind.DECLARED) {
+      return false;
+    }
+
+    DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+    AnnotatedDeclaredType uncapturedType = (AnnotatedDeclaredType) type;
```

Comment:
Since the uncaptured type is the first formal parameter, I would swap the order of this assignment and the previous one.  I think that makes the code a bit more uniform and easier to read.  I made similar changes in a number of other places in the code.